### DISCUSSION TITLE
Auth Go Documentation

### DISCUSSION THREAD

**Discussion by Zwnow** [pocketbase/pocketbase#5694] [Q&A] [answered]
Hello everyone,

unfortunately I was not able to find anything (quickly) regarding my question.

I am looking for documentation on how to verify if the Authorization Token is valid in backend in Go?

The only auth related docs I was able to find are the Dart and Javascript ones, but they seem to be frontend related? My Go app does not have an authStore. I need a way to verify and generate the tokens in my backend. Or does pocketbase do all that automatically?
_____________________________
~~Edit:~~

~~I also can't fill the Password & PasswordConfirm attributes in the following example:~~

~~The req.FormValues contain data, but the form doesn't only in these fields.~~
I am blind and did not see password & passwordConfirm were lowercased

**Comment by ganigeorgiev** [comment, marked as answer]
It is not clear what you are asking or trying to do.

There is no other documentation than what is available on the website and the [godoc](https://pkg.go.dev/github.com/pocketbase/pocketbase) of the package.

For retrieving the auth state from inside a route, see https://pocketbase.io/docs/go-routing/#retrieving-the-current-auth-state.
For manually generating tokens, see the [`tokens` subpackage](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.21/tokens).
For manually validating tokens, see the [`Dao.FindAuthRecordByToken`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.21/daos#Dao.FindAuthRecordByToken).

For example:

```go
// manually generating auth record
token, err := tokens.NewRecordAuthToken(app, authRecord)

// manually verifying auth JWT and retrieving its related auth record 
authRecord, err := app.Dao().FindAuthRecordByToken("TOKEN", app.Settings().RecordAuthToken.Secret)
```

_Note that all of the above has slightly changed and simplified with v0.23.0 (it is in the `develop` branch)._


With that said, if you are trying to create Go SSR apps with PocketBase then I'd advice against it, especially if you are new to Go, because we don't have the necessary helpers, or the proper guides/documentation as you've already found out, for this type of apps at the moment and if you don't feel comfortable reading the PocketBase source code then you'll be disappointed and waste your time (see also https://github.com/pocketbase/pocketbase/discussions/5307). And again, with v0.23.0 (_there are no ETAs yet_) most of the Go apis have changed due to the new hooks and router so even if you create your own custom helpers and utilities you'll have to rewrite them if you decide later to support newer PocketBase versions.

**Reply by Zwnow** [reply]
Hmm I see, yea I wanted to create a SSR app with Pocketbase as it seemed like a great tool. Too bad it's not meant for this.

**Reply by DanyWallace** [reply]
it's pretty good but not for people who lack experience, imho it's no different than using the standard http library in go lol

**Reply by Zwnow** [reply]
Then why recommend not using it for SSR apps? How am I supposed to gain experience?
